internal/cloudapi/v2/v2.go

```diff
@@ -92,6 +112,30 @@ func (s *Server) IncRequests(next echo.HandlerFunc) echo.HandlerFunc {
 	}
 }
 
+func (s *Server) ValidateRequest(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		request := c.Request()
+
+		route, params, err := s.router.FindRoute(request)
+		if err != nil {
+			return err
+		}
+
+		input := &openapi3filter.RequestValidationInput{
+			Request:    request,
+			PathParams: params,
+			Route:      route,
+		}
+
+		ctx := request.Context()
+		if err := openapi3filter.ValidateRequest(ctx, input); err != nil {
+			return err
```

Comment:
Same here, but I wonder if it's possible we can get 'why' it's a 400 from the error here. The advantage of checking stuff in the handlers themselves is that we can return different self-defined error codes. If we can get a single useful string maybe we could attach it to the return error message.